api/src/main/java/me/lokka30/treasury/api/common/response/Subscriber.java

```diff
@@ -0,0 +1,109 @@
+package me.lokka30.treasury.api.common.response;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An interface accepting responses from a provider.
+ * Used to subscribe to a request that will be completed at
+ * some point in the future.
+ * Example usage see class level
+ * javadoc of {@link me.lokka30.treasury.api.economy.response.EconomySubscriber}
+ *
+ * @param <T> the type of value expected on success
+ * @author Jikoo
+ * @since v1.1
+ */
+public interface Subscriber<T, E extends TreasuryException> {
+
+    /**
+     * Respond to the subscriber with a successful invocation.
+     *
+     * @param t the value of the successful invocation
+     * @author Jikoo
+     * @since v1.1
+     */
+    void succeed(@NotNull T t);
+
+    /**
+     * Respond to the subscriber with an invocation failure.
+     *
+     * @param exception an {@link E} detailing the reason for failure
+     * @author Jikoo
+     * @since v1.1
```

Comment:
`@since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}`

OR

`@since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_1 v1.1}`